float32[3] right_vel
float32[3] left_vel